### ISSUE TITLE
Dowloading and Reconstructing Conversation Threads

### ISSUE THREAD

**Issue by mcolemann** [DocNow/twarc#523]
Hi!

I have a lot of tweet ids (33 millions). I would like to download the content of the tweets and reconstruct the conversation threads (to do a sentiment analysis). I used Hydrator to download the content of the tweets and now I have 33 .csv files with the information about the tweets (text,...).

Is there a way to reconstruct the conversation threads in the .csv files using twarc? Or is it better to do everything with twarc (also the download of the tweets)?
Would it take much time to redownload everything with twarc? Is there a limit in the download?

Thanks a lot!

**Comment by edsu** [comment]
Currently the Hydrator is using Twitter's old v1.1 API. Twitter's new v2 API returns new information like `reply_count` and `conversation_id` which I think will be critical for your project. `reply_count` tells you how many replies have been made to a tweet, and `conversation_id` is a unique identifier for the thread that a tweet might be a part of. These two pieces of information are important because they would allow you to see what conversation threads exist in your hydrated data.

Since twarc can talk to the v2 API If you use it to hydrate your data you will get this additional information, and then can filter your hydrated data looking for conversation_ids of interest, and then got get the full conversation threads for them.

The problem is, you will need Academic Research Product Track access to search by `conversation_id` for tweets outside of the 7 day window that the standard v2 API provides. So definitely apply for that if you study or work at an academic institution. Also, the Academic Track has a 10 million / month quota (instead of the standard 100,000), which you will surely run up against with this volume of data.

Let me know if you want to chat about it sometime or drop into the [DocNow Slack](https://bit.ly/docow-slack) if you want to chat more there. I'm actually working on a very similar process at the moment, doing sentiment analysis of retrieved threads, so it would be fun to compare notes!

**Comment by SamHames** [comment]
Just to add on to Ed's comment though, if you already have the tweet ids of
interest you won't need academic access to hydrate them - that works the
same in the V2 API as the V1.1.

It may not be possible to reconstruct the complete threads from V1.1
either, as deleted tweets will break the chain of replies.

**Comment by edsu** [comment]
Yes, good point about hydration not counting against the 10M/100K limit. I think it's a good reason to update Hydrator to use the v2 API.
